### ISSUE TITLE
Principled IOR mismatch

### ISSUE THREAD

**Issue by PearCoding** [PearCoding/Ignis#16]
The ior from the principled bsdf (aka. disney bsdf) does not work for all ior values. IOR=1 returns just black

**Comment by pgrit** [comment]
IOR=1 is a NaN causing corner-case for microfacet distributions. Because there's no such thing as a rough yet non-existent interface between two media ;)

You could add a special case that switches to a non-rough dielectric, or simply enforce a lower bound of 1.00001 or sth.

**Comment by PearCoding** [comment]
True, but the behavior is inconsistent in the implementation. The rough dielectric bsdf has no problem dealing with it... the principled bsdf however does. So there is a "happy accident" somewhere
